core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -178,7 +178,7 @@ private Translog openTranslog(EngineConfig engineConfig, IndexWriter writer, boo
             }
         }
         final Translog translog = new Translog(translogConfig);
-        if (generation == null) {
+        if (generation == null || generation.translogUUID == null) {
             logger.debug("no translog ID present in the current generation - creating one");
```

Comment:
can we change this log to indicated why we ended up here? (now it says only translog ID)